ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutor.java

```diff
@@ -223,13 +229,21 @@ public PullQueryResult execute(
           contextStacker,
           pullQueryMetrics);
 
-      return handlePullQuery(
+      final PullQueryResult result = handlePullQuery(
           statement,
           executionContext,
           serviceContext,
           pullQueryContext,
           routingOptions
       );
+
```

Comment:
There was a reason for these measurements to be in the `StreamedQueryResource`. They measure a more realistic latency that accounts also for query parsing etc. , basically all that happens before the request reaches the `PullQueryExecutor`.

Reply:
+1

Reply:
My original thinking was that such CPU bound things like parsing should be an order or magnitude or two faster than IO like reading from disk or network.  I do recall that in earlier benchmarking, we didn't seem to be IO bound, but maybe that's probably a separate issue.

If what we really want is latency for the whole endpoint, which I agree is preferable, it seems like it would be good to have a more generic latency metric for all resources -- that way we don't special case this.  Not sure if that's possible with Vert.x, but I imagine it is.  So as to minimize this PR and keep consistency with existing metric naming, I'll stick to the pull query use case for now.  Ok, I'll change this part back and add in the web socket.

Reply:
That's a good point - might be worth asking @purplefox and/or googling about vert.x to see if there is a generic way to get metrics on _all_ requests.  I know this is possible in Jetty, but don't know vert.x.